src/dsa/rpo_stark/stark/prover.rs

```diff
@@ -0,0 +1,145 @@
+use core::marker::PhantomData;
+
+use rand_chacha::ChaCha20Rng;
+use winter_air::ZkParameters;
+use winter_crypto::{ElementHasher, SaltedMerkleTree};
+use winter_math::{fields::f64::BaseElement, FieldElement};
+use winter_prover::{
+    matrix::ColMatrix, DefaultConstraintEvaluator, DefaultTraceLde, ProofOptions, Prover,
+    StarkDomain, Trace, TraceInfo, TracePolyTable, TraceTable,
+};
+use winter_utils::{Deserializable, Serializable};
+use winterfell::{
+    AuxRandElements, CompositionPoly, CompositionPolyTrace, ConstraintCompositionCoefficients,
+    DefaultConstraintCommitment, PartitionOptions,
+};
+
+use super::air::{PublicInputs, RescueAir, HASH_CYCLE_LEN};
+use crate::{hash::rpo::Rpo256, rand::RpoRandomCoin, Word, ZERO};
+
+// PROVER
+// ================================================================================================
+
+pub struct RpoSignatureProver<H: ElementHasher>
+where
+    H: Sync,
+{
```

Comment:
A few nits:
- Since `H` is a generic parameter, this is not just for RPO, right? Should we rename this into just `SignatureProver`?
- I don't think we need to have a `where` clause here (could just do `+ Sync` inline).
- Let's add some doc comments to this struct.
- Could this struct be `pub(super)` or `pub(crate)`?

Reply:
1. RPO is to refer to the one-way problem used to define the signature which is different than the hash function used in the STARK. Added a comment to explain this but also open to remove the prefix altogether.
2. Changed
3. Added
4. Indeed, fixed now